ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -234,3 +239,75 @@ message DisclosedContract {
   // Required
   ContractMetadata metadata = 4;
 }
+
+message TransferCommand {
+  // Identifier of the on-ledger workflow that this command is a part of.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 1;
+
+  // Uniquely identifies the application or participant user that issued the command.
+  // Must be a valid ApplicationIdString (as described in ``value.proto``).
+  // Required unless authentication is used with a user token or a custom token specifying an application-id.
+  // In that case, the token's user-id, respectively application-id, will be used for the request's application_id.
+  string application_id = 2;
+
+  // Uniquely identifies the command.
+  // The triple (application_id, submitter, command_id) constitutes the change ID for the intended ledger change.
+  // The change ID can be used for matching the intended ledger changes with all their completions.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string command_id = 3;
+
+  // Party on whose behalf the command should be executed.
+  // If ledger API authorization is enabled, then the authorization metadata must authorize the sender of the request
+  // to act on behalf of the given party.
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string submitter = 4;
+
+  oneof command {
+    TransferOutCommand transfer_out_command = 5;
+    TransferInCommand transfer_in_command = 6;
+  }
+
+  // If omitted, the participant or the committer may set a value of their choice.
+  // Optional
```

Comment:
Why is this one optional ?

Reply:
for the same reason it is optional for the submission_id for the daml commands. I guess to let client have an option not worrying about setting it correctly.
There was more comment on this for the other submission_id, I moved them over to this too: it is applicable and explains more clearly what is going on here.